server/src/main/java/org/apache/iotdb/db/engine/storagegroup/StorageGroupProcessor.java

```diff
@@ -2410,4 +2410,58 @@ public boolean isFileAlreadyExist(TsFileResource tsFileResource, long partitionN
 
     void call(TsFileResource caller);
   }
+
+  /**
+   * remove all partitions that satisfy a filter.
+   * @param filter
+   */
+  public void removePartitions(TimePartitionFilter filter) {
+    // this requires blocking all other activities
+    insertLock.writeLock().lock();
+    mergeLock.writeLock().lock();
+    try {
+      // abort ongoing merges
+      MergeManager.getINSTANCE().abortMerge(storageGroupName);
```

Comment:
I notice you add `storageGroupSubTasks` and `storageGroupMainTasks` in 6.9. When to clear the finished tasks?

Reply:
There is a TaskCleanerThread in MergeManager.